### PR TITLE
(plugin) apps::monitoring::kadiska - add new plugin for Kadiska

### PR DESCRIPTION
**Pull request by lchrdn**
Adding a new plugin for Kadiska monitoring application that helps you retrieve information from Kadiska Platform.

### CODE REVIEW

File: apps/monitoring/kadiska/mode/watcherstatistics.pm [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+
+sub set_counters {
+    my ($self, %options) = @_;
+
+    $self->{maps_counters_type} = [
+        { name => 'watchers', type => 1, cb_prefix_output => 'prefix_output', message_multiple => 'All Watchers are OK' }
+    ];
+
+    $self->{maps_counters}->{watchers} = [
+        { label => 'errors-prct', nlabel => 'errors.persecond.percentage', set => {
```
Comment (lchrdn) [inline apps/monitoring/kadiska/mode/watcherstatistics.pm:43]: nlabel => watcher.errors.percentage

File: apps/monitoring/kadiska/mode/watcherstatistics.pm [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+                key_values => [ { name => 'errors_prct' }, { name => 'display' } ],
```
Comment (lchrdn) [inline apps/monitoring/kadiska/mode/watcherstatistics.pm:44]:
```suggestion
                key_values => [ { name => 'errors_prct' } ],
```

File: apps/monitoring/kadiska/mode/watcherstatistics.pm [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+                output_template => 'errors: %.2f%%',
+                perfdatas => [
+                    { label => 'errors_prct', template => '%.2f', unit => '%', min => 0, max => 100, label_extra_instance => 1, instance_use => 'display' },
```
Comment (lchrdn) [inline apps/monitoring/kadiska/mode/watcherstatistics.pm:47]:
```suggestion
                    { label => 'errors_prct', template => '%.2f', unit => '%', min => 0, max => 100, label_extra_instance => 1},
```

File: apps/monitoring/kadiska/mode/watcherstatistics.pm [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+use base qw(centreon::plugins::templates::counter);
+
+use strict;
+use warnings;
+use centreon::plugins::templates::catalog_functions qw(catalog_status_threshold_ng);
+
+sub prefix_output {
+    my ($self, %options) = @_;
+    
+    return "Watcher '" . $options{instance_value}->{display} . "' ";
```
Comment (lchrdn) [inline apps/monitoring/kadiska/mode/watcherstatistics.pm:32]:
```suggestion
    return "Watcher '" . $options{instance} . "' ";
```

Comment (lchrdn) [reply apps/monitoring/kadiska/mode/watcherstatistics.pm:32]: a verif

File: apps/monitoring/kadiska/mode/watcherstatistics.pm [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+    );
+
+    $self->{watchers} = {};
+    foreach my $watcher (@{$results->{data}}) {
+        next if (defined($self->{option_results}->{filter_watcher_name}) && $self->{option_results}->{filter_watcher_name} ne ''
+            && $watcher->{'watcher_id:group'} !~ /$self->{option_results}->{filter_watcher_name}/);
+
+        my $instance = $watcher->{'watcher_id:group'};
+
+        $self->{watchers}->{$instance} = { display => $instance,
```
Comment (lchrdn) [inline apps/monitoring/kadiska/mode/watcherstatistics.pm:143]:
```suggestion
        $self->{watchers}->{$instance} = {  
```

File: apps/monitoring/kadiska/custom/api.pm [hunk 6]
```diff
@@ -0,0 +1,284 @@
...
+    }
+
+    $self->settings(environment_header => 1, organization_header => 1);
+
+    my $encoded_form_post;
+
+    if (defined($options{query_form_post})) {
+
+        my $end = time() * 1000;
+        my $begin = ($end - (86400 * $self->{period} * 1000));
```
Comment (lchrdn) [inline apps/monitoring/kadiska/custom/api.pm:182]: put minutes instead of days

**Review by Sims24 — APPROVED** [review]

**Pull request merged**